validity/src/prom.rs

```diff
@@ -45,9 +45,59 @@ pub fn custom_gauges() {
         "succinct_min_block_to_prove_to_agg",
         "Minimum block number required to prove for aggregation"
     );
+    describe_gauge!(
+        "succinct_proof_request_retry_count",
+        "Number of proof request retries"
+    );
 
     // Error gauges
-    describe_gauge!("succinct_error_count", "Number of errors");
+    describe_gauge!("succinct_total_error_count", "Number of total errors");
+    describe_gauge!(
+        "succinct_proof_request_timeout_error_count",
+        "Number of proof request timeout errors"
+    );
+    describe_gauge!("succinct_retry_error_count", "Number of retry errors");
+    describe_gauge!(
+        "succinct_witnessgen_error_count",
+        "Number of witness generation errors"
+    );
+    describe_gauge!(
+        "succinct_execution_error_count",
+        "Number of execution errors"
+    );
+    describe_gauge!(
+        "succinct_mock_range_proof_request_error_count",
+        "Number of mock range proof request errors"
+    );
+    describe_gauge!(
+        "succinct_mock_agg_proof_request_error_count",
+        "Number of mock aggregation proof request errors"
+    );
+    describe_gauge!(
+        "succinct_range_proof_request_error_count",
+        "Number of range proof request errors"
+    );
+    describe_gauge!(
+        "succinct_agg_proof_request_error_count",
+        "Number of aggregation proof request errors"
+    );
+    describe_gauge!(
+        "succinct_relay_agg_proof_error_count",
+        "Number of relay aggregation proof errors"
+    );
+}
+
+pub fn init_gauges() {
+    gauge!("succinct_total_error_count").set(0.0);
+    gauge!("succinct_proof_request_timeout_error_count").set(0.0);
+    gauge!("succinct_retry_error_count").set(0.0);
+    gauge!("succinct_witnessgen_error_count").set(0.0);
+    gauge!("succinct_execution_error_count").set(0.0);
+    gauge!("succinct_mock_range_proof_request_error_count").set(0.0);
+    gauge!("succinct_mock_agg_proof_request_error_count").set(0.0);
+    gauge!("succinct_range_proof_request_error_count").set(0.0);
+    gauge!("succinct_agg_proof_request_error_count").set(0.0);
+    gauge!("succinct_relay_agg_proof_error_count").set(0.0);
```

Comment:
Can you make an enum or something similar so we have better type-safety on the gauges?

For example here, in `custom_gauges` you describe `succinct_proof_request_retry_count`, but you never init this gauge.

There's probably a programmatic way to do it, that will make this easier to parse too.